src/proposals/20241114/SparkEthereum_20241114.t.sol

```diff
@@ -0,0 +1,347 @@
+
+        executePayload(payload);
+
+        assertEq(almProxy.hasRole(almProxy.CONTROLLER(), Ethereum.ALM_CONTROLLER),     true, "incorrect-controller-almProxy");
+        assertEq(rateLimits.hasRole(rateLimits.CONTROLLER(), Ethereum.ALM_CONTROLLER), true, "incorrect-controller-rateLimits");
+        assertEq(controller.hasRole(controller.FREEZER(), FREEZER),                    true, "incorrect-freezer-controller");
+        assertEq(controller.hasRole(controller.RELAYER(), RELAYER),                    true, "incorrect-relayer-controller");
+
+        _assertRateLimit(controller.LIMIT_USDC_TO_CCTP(), type(uint256).max, 0);
+        _assertRateLimit(
+            RateLimitHelpers.makeDomainKey(controller.LIMIT_USDC_TO_DOMAIN(), CCTPForwarder.DOMAIN_ID_CIRCLE_BASE),
+            4_000_000e6,
+            2_000_000e6 / uint256(1 days)
+        );
+        _assertRateLimit(controller.LIMIT_USDS_MINT(),    4_000_000e18, 2_000_000e18 / uint256(1 days));
+        _assertRateLimit(controller.LIMIT_USDS_TO_USDC(), 4_000_000e6,  2_000_000e6 / uint256(1 days));
+
+        assertEq(controller.mintRecipients(CCTPForwarder.DOMAIN_ID_CIRCLE_BASE), bytes32(uint256(uint160(Base.ALM_PROXY))));
+    }
+
+    function _setupCrossChainTest() internal {
+        mainnet.selectFork();
+        executePayload(payload);
+        // TODO use the actual cross-chain message to execute when deployed
+        base.selectFork();
+        executePayloadBase(basePayload);
+    }
+
+    function testDepositUSDSandSUSDSPSM3() public {
+        _setupCrossChainTest();
+
+        mainnet.selectFork();
+
+        uint256 susdsShares = IERC4626(Ethereum.SUSDS).convertToShares(SUSDS_DEPOSIT_AMOUNT);
+
+        base.selectFork();
+
+        assertEq(IERC20(Base.USDS).balanceOf(Base.ALM_PROXY),  0);
+        assertEq(IERC20(Base.SUSDS).balanceOf(Base.ALM_PROXY), 0);
+
+        OptimismBridgeTesting.relayMessagesToDestination(nativeBridge, true);
+
+        assertEq(IERC20(Base.USDS).balanceOf(Base.ALM_PROXY),  USDS_BRIDGE_AMOUNT);
+        assertEq(IERC20(Base.SUSDS).balanceOf(Base.ALM_PROXY), susdsShares);
+
+        // Deposit USDS and sUSDS
+        assertEq(IERC20(Base.USDS).balanceOf(Base.ALM_PROXY),  USDS_BRIDGE_AMOUNT);
+        assertEq(IERC20(Base.SUSDS).balanceOf(Base.ALM_PROXY), susdsShares);
+        assertEq(IERC20(Base.USDS).balanceOf(Base.PSM3),       0);
+        assertEq(IERC20(Base.SUSDS).balanceOf(Base.PSM3),      0);
+
+        vm.startPrank(RELAYER);
+        ForeignController(Base.ALM_CONTROLLER).depositPSM(Base.USDS,  USDS_BRIDGE_AMOUNT);
+        ForeignController(Base.ALM_CONTROLLER).depositPSM(Base.SUSDS, susdsShares);
+        vm.stopPrank();
+
+        assertEq(IERC20(Base.USDS).balanceOf(Base.ALM_PROXY),  0);
+        assertEq(IERC20(Base.SUSDS).balanceOf(Base.ALM_PROXY), 0);
+        assertEq(IERC20(Base.USDS).balanceOf(Base.PSM3),       USDS_BRIDGE_AMOUNT);
+        assertEq(IERC20(Base.SUSDS).balanceOf(Base.PSM3),      susdsShares);
+    }
```

Comment:
```suggestion
    function testDepositUSDSandSUSDSPSM3() public {
        _setupCrossChainTest();

        mainnet.selectFork();

        uint256 mainnetTimestamp = block.timestamp;

        uint256 susdsShares = IERC4626(Ethereum.SUSDS).convertToShares(SUSDS_DEPOSIT_AMOUNT);

        vm.warp(mainnetTimestamp + 1 days);

        uint256 mainnetSUsdsAssets = IERC4626(Ethereum.SUSDS).convertToAssets(susdsShares);

        base.selectFork();

        vm.warp(mainnetTimestamp + 1 days);

        uint256 susdsPsmShares  = IPSMLike(Base.PSM3).convertToShares(Base.SUSDS, susdsShares);
        uint256 baseSUsdsAssets = IPSMLike(Base.PSM3).convertToAssetValue(susdsPsmShares);

        // Ensure cross-chain accounting is correct at the same timestamp
        assertEq(mainnetSUsdsAssets, baseSUsdsAssets);

        assertEq(IERC20(Base.USDS).balanceOf(Base.ALM_PROXY),  0);
        assertEq(IERC20(Base.SUSDS).balanceOf(Base.ALM_PROXY), 0);

        OptimismBridgeTesting.relayMessagesToDestination(nativeBridge, true);

        assertEq(IERC20(Base.USDS).balanceOf(Base.ALM_PROXY),  USDS_BRIDGE_AMOUNT);
        assertEq(IERC20(Base.SUSDS).balanceOf(Base.ALM_PROXY), susdsShares);

        // Deposit USDS and sUSDS
        assertEq(IERC20(Base.USDS).balanceOf(Base.ALM_PROXY),  USDS_BRIDGE_AMOUNT);
        assertEq(IERC20(Base.SUSDS).balanceOf(Base.ALM_PROXY), susdsShares);
        assertEq(IERC20(Base.USDS).balanceOf(Base.PSM3),       0);
        assertEq(IERC20(Base.SUSDS).balanceOf(Base.PSM3),      0);

        assertEq(IPSMLike(Base.PSM3).shares(Base.ALM_PROXY),  0);

        vm.startPrank(RELAYER);
        ForeignController(Base.ALM_CONTROLLER).depositPSM(Base.USDS,  USDS_BRIDGE_AMOUNT);
        ForeignController(Base.ALM_CONTROLLER).depositPSM(Base.SUSDS, susdsShares);
        vm.stopPrank();

        assertEq(IERC20(Base.USDS).balanceOf(Base.ALM_PROXY),  0);
        assertEq(IERC20(Base.SUSDS).balanceOf(Base.ALM_PROXY), 0);
        assertEq(IERC20(Base.USDS).balanceOf(Base.PSM3),       USDS_BRIDGE_AMOUNT);
        assertEq(IERC20(Base.SUSDS).balanceOf(Base.PSM3),      susdsShares);

        // Ensure that value controlled by the ALM proxy in the PSM is equal to total value deposited
        uint256 proxyPsmShares = IPSMLike(Base.PSM3).shares(Base.ALM_PROXY);

        assertEq(proxyPsmShares, susdsPsmShares + 1_000_000e18);

        assertEq(
            IPSMLike(Base.PSM3).convertToAssetValue(proxyPsmShares),
            mainnetSUsdsAssets + 1_000_000e18
        );
    }
```

Reply:
Adding some assertions to check cross chain accounting works and that value controlled by proxy after PSM deposit is whats expected

Reply:
This is not a great diff. I copy and paste and all the formatting is broken.